src/open_inwoner/components/templates/components/Header/Header.html

```diff
@@ -11,43 +11,103 @@
 
             <dialog class="header__submenu">
                 <nav class="header__actions" aria-label="Zoek navigatie mobiel">
+                    {% url 'search:search' as search_url %}
+                    {% render_form form=search_form method="GET" form_action=search_url inline=True spaceless=False %}
+                        {% input search_form.query no_label=True %}
+                        {% form_actions primary_icon="search" primary_text=_("Zoeken") hide_primary_text=True %}
+                    {% endrender_form %}
+                </nav>
+
+                <nav class="primary-navigation" aria-label="Hoofd navigatie">
+                    <ul class="primary-navigation__list">
+                        <li class="primary-navigation__list-item">
+                            {% trans "Home" as link_text %}
+                            {% url 'root' as link %}
+                            {% link text=link_text href=link icon="apps" icon_position="before" %}
+                        </li>
+                        <li class="primary-navigation__list-item dropdown-nav__toggle">
+                            <a href="#" class="link link--toggle link--icon link--icon-position-before" aria-label="Thema's" title="Thema's">
+                                <span >{% trans "Themes" %}</span>
+                                <span aria-hidden="true" class="material-icons-outlined ">description</span>
+                                {% icon icon="expand_more" icon_position="after" icon_outlined=True %}
+                            </a>
+
+                            {% if categories %}
+                                <ul class="primary-navigation__list subpage-list">
+                                    {% for category in categories %}
+                                        <li class="primary-navigation__list-item">
+                                            {% url 'pdc:category_detail' slug=category.slug as category_href %}
+                                            {% link text=category.name href=category_href %}
+                                        </li>
+                                    {% endfor %}
+                                </ul>
+                            {% endif %}
+                        </li>
+
+                        {% if request.user.is_authenticated %}
+                            <li class="primary-navigation__list-item">
+                                {% link text=_("Mijn profiel") href='accounts:my_profile' icon="inventory_2" icon_position="before" icon_outlined=True %}
+                            </li>
+                            <li class="primary-navigation__list-item">
+                                {% link text=_('Mijn berichten') href='accounts:inbox' icon="inbox" icon_position="before" %}
+                                {% if request.user.get_new_messages_total %}
+                                    {% with "("|addstr:request.user.get_new_messages_total|addstr:")" as message_total %}
```

Comment:
Pretty impressive filter use :smile: 

The `request.user.get_new_messages_total` is doing queries though, so it is better to put its result into a variable and re-use it, instead of querying the same thing twice.

Reply:
@Bartvaderkin  I copied all of the already existing code from the PrimaryNavigation template to the Header template, so I did not come up with these myself; this will be a learning experience for me!

Reply:
With the help of Vasileios I added `{% with request.user.get_new_messages_total as total_messages %}` in both header and Primary navigation.